proto/ignition/msgs/statistic.proto

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package ignition.msgs;
+option java_package = "com.ignition.msgs";
+option java_outer_classname = "StatisticProtos";
+
+/// \ingroup ignition.msgs
+/// \interface Statistic
+/// \brief A message that contains statistics data.
+
+import "ignition/msgs/header.proto";
+
+message Statistic
+{
+  enum DataType
+  {
+    UNITIALIZED = 0;
+    AVERAGE      = 1;
+    MINIMUM      = 2;
+    MAXIMUM      = 3;
+    STDDEV       = 4;
+    SAMPLE_COUNT = 5;
+  }
+
+  /// \brief Optional header data
+  Header header  = 1;
+
+  DataType type  = 2;
+
+  string   name  = 3;
+
+  double   value = 4;
+}
+
+/// \brief A named group of statistics.
+message StatisticsGroup
+{
+  /// \brief Optional header data.
+  Header header                 = 1;
+
+  /// \brief Name of the group.
+  string name                   = 2;
+
+  /// \brief Statistics the belong to this group.
+  repeated Statistic statistics = 3;
+}
+
+message Metric
+{
+  /// \brief Optional header data
+  Header header  = 1;
+
+  /// \brief Unit of measurement such as seconds, meters, liters.
+  string   unit  = 2;
+
+  /// \brief Zero or more named groups of statistics. A statistic group is
+  /// used to bundle data into a logical set with an associated name.
+  repeated StatisticsGroup statistics_groups = 3;
+
+  /// \brief Zero or more statistics.
+  repeated Statistic statistics          = 4;
```

Comment:
Do we really need this field? Can't we achieve the same result by adding one `statistics_group`, and then, one single statistics inside?

Reply:
A group is a logical organization of data. For example, all statistics related to transport publication. 

There are cases where a single statistics makes sense to live on its own. 

Since these are proto messages, each is optional.

Reply:
Sounds good, thanks for the clarification.